### PR TITLE
Migrate Pin to Error-v2

### LINKED ISSUE #1134: Error 2.0: Migrate Pin to Returns<T>

### PR DESCRIPTION
**Pull request by HuangDave**
Resolves #1134

### CODE REVIEW

**Comment by kammce** [comment]
This looks great, but I fear that making this change to integral parts of the code base that people may be using before summer is a bad idea. I made this change to storage because it was a new interface made during the academic school year. For now, I can review these and we can discuss them in the PR, but this MUST NOT be merged into the main code base.

File: library/L1_Peripheral/pin.hpp [hunk 1]
```diff
@@ -93,26 +98,29 @@ class Pin
   /// the job of the L1 peripheral that uses this pin to manage its own function
   /// code usage.
   ///
-  /// @param function - pin function code
+  /// @param function pin function code
   virtual void SetPinFunction(uint8_t function) const = 0;
```
Comment (kammce) [inline library/L1_Peripheral/pin.hpp:102]: I feel like this could also return an error, do you agree?

File: library/L1_Peripheral/pin.hpp [hunk 2]
```diff
@@ -93,26 +98,29 @@ class Pin
...
+
   /// Set pin's resistor pull, setting ot either no resistor pull, pull down,
   /// pull up and repeater.
   ///
   /// @param resistor - which resistor setup you would like.
   virtual void SetPull(Resistor resistor) const = 0;
```
Comment (kammce) [inline library/L1_Peripheral/pin.hpp:108]: Couldn't this return an error if the resistor state desired does not exist for that platform?

File: library/L1_Peripheral/pin.hpp (hunk 1 above)
Comment (HuangDave) [reply library/L1_Peripheral/pin.hpp:102]: I agree, we could check for whether the function code is valid or not.

**Comment by kammce** [comment]
Closing this for now.

**Pull request merged**